### DISCUSSION TITLE
how to start read skywalking ?(with java-agent)

### DISCUSSION THREAD

**Discussion by heihei180** [apache/skywalking#8031] [Q&A] [answered]
when i cloned the `https://github.com/apache/skywalking.git` , i can't find where java-agent source code，after a long time，Stumbled upon this: `https://github.com/apache/skywalking-java` ,, That's exciting， I remember they were on the same project before，Are we separated now? Or I didn't find it right？if true, can u give me some guide ? i want to read `skywalking` source code again~ , with java agent. tks~

**Comment by wu-sheng** [comment, marked as answer]
Look at here for more context, https://github.com/apache/skywalking/discussions/7548.

**Reply by heihei180** [reply]
thank u very much !
